com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -288,7 +288,35 @@ private void doScan(IClassScanner scanner, Set<String> classNames) {
         }
     }
 
+    static String[][] extensionMapping = new String[][] {
+        {".camera", ".camerac"},
+        {".buffer", ".bufferc"},
+        {".mesh", ".meshc"},
+        {".collectionproxy", ".collectionproxyc"},
+        {".collisionobject", ".collisionobjectc"},
+        {".emitter", ".emitterc"},
+        {".particlefx", ".particlefxc"},
+        {".gui", ".guic"},
+        {".model", ".modelc"},
+        {".script", ".scriptc"},
+        {".sound", ".soundc"},
+        {".wav", ".soundc"},
+        {".collectionfactory", ".collectionfactoryc"},
+        {".factory", ".factoryc"},
+        {".light", ".lightc"},
+        {".label", ".labelc"},
+        {".sprite", ".spritec"},
+        {".tilegrid", ".tilemapc"},
+        {".tilemap", ".tilemapc"},
+    };
+
     public String replaceExt(String inExt) {
+        for (int i = 0; i < extensionMapping.length; i++) {
+            if (extensionMapping[i][0].equals(inExt))
+            {
+                return extensionMapping[i][1];
+            }
+        }
```

Comment:
@mathiaswking  I iterate extension mappings from `GameObjectBuilder.java` and return right extension here now